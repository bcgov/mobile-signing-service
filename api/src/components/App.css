.container {
  width: 80%;
  max-width: 1200px;
  margin: 3em auto;
  padding: 1em;
  border-style: solid;
  border-width: 1px;
  border-color: lightgray;
}

.container * {
  box-sizing: border-box;
}

.flex-outer,
.flex-inner {
  list-style-type: none;
  padding: 0;
}

.flex-outer {
  max-width: 85%;
  margin: 0 auto;
}

.flex-outer li,
.flex-inner {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}

.flex-inner {
  padding: 0 10px;
  justify-content: space-between;
}

.flex-outer > li:not(:last-child) {
  margin-bottom: 20px;
}

.flex-outer li label,
.flex-outer li p {
  padding: 8px;
  font-weight: 300;
}

.flex-outer > li > label,
.flex-outer li p {
  flex: 1 0 120px;
  max-width: 320px;
  margin-right: 20px;
}

.flex-outer > li > label + *,
.flex-inner {
  flex: 1 0 220px;
}

.flex-outer li input,
.flex-outer li textarea {
  padding: 15px;
  border: none;
  margin-bottom: .5rem;
}

.flex-outer li button {
  position: relative;
  margin-left: auto;
  padding: 8px 16px;
  border: none;
  background-color: #fcba19;
  color: #003366;
  text-transform: uppercase;
  letter-spacing: 0.2em;
  border-radius: 2px;
  outline-width: 0;
  text-decoration: none;
  overflow: hidden;
  cursor: pointer;
  -webkit-transition-duration: 0.4s; /* Safari */
  transition-duration: 0.4s;
}

.flex-outer li button:after {
  content: '';
  background: #003366;
  display: block;
  position: absolute;
  padding-top: 300%;
  padding-left: 350%;
  margin-left: -20px !important;
  margin-top: -120%;
  opacity: 0;
  transition: all 0.8s;
}

.flex-outer li button:active:after {
  padding: 0;
  margin: 0;
  opacity: 1;
  transition: 0s;
}

.flex-inner li {
  width: 100px;
}

.file-download-icon {
  color: #fcba19;
}

a {
  color: #003366;
  /* text-transform: uppercase; */
  /* letter-spacing: 0.05em; */
  text-decoration: none;
}

button {
  font-size: 12px;
}
