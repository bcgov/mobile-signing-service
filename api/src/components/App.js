import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createSigningJob } from '../actionCreators';
import { authenticateFailed, authenticateSuccess } from '../actions';
import implicitAuthManager from '../auth';
import { AC_ROLE } from '../constants';
import './App.css';
import FileUpload from './FileUpload/FileUpload';
import Instruction from './Instruction/Instruction';
import JobStatusIndicator from './JobStatusIndicator/JobStatusIndicator';
import Footer from './UI/Footer';
import Header from './UI/Header';

export class App extends Component {
  state = {
    loading: true,
    userAgreedToTerms: false,
  };

  componentDidMount = () => {
    implicitAuthManager.registerHooks({
      onAuthenticateSuccess: () => this.props.login(),
      onAuthenticateFail: () => this.props.logout(),
      // onAuthLocalStorageCleared: () => this.props.logout(),
    });
    // don't call function if on localhost
    if (!window.location.host.match(/localhost/)) {
      implicitAuthManager.handleOnPageLoad();
    }
  };

  onUserDoesAgree = e => {
    this.setState({ userAgreedToTerms: !this.state.userAgreedToTerms });
  };

  onPlatformChanged = e => {
    this.setState({ platform: e.currentTarget.value });
  };

  onFileAccepted = files => {
    this.setState({ files });
  };

  validateForm = () => {
    if (!implicitAuthManager.isAuthenticated()) {
      alert('You need to login before you can submit signing jobs.');
      return false;
    }

    if (!implicitAuthManager.roles.includes(AC_ROLE)) {
      alert('You need the proper role to be able to create signing jobs');
      return false;
    }

    if (!this.state.files) {
      alert('You need to add a file to be uploaded for signing.');
      return false;
    }

    if (!this.state.platform) {
      alert('You need to select a platform for your signing job.');
      return false;
    }

    if (!this.state.userAgreedToTerms) {
      alert('You need to agree to complete a STRA & PIA before you can submit signing jobs.');
      return false;
    }

    return true;
  };

  render() {
    return (
      <div>
        <Header authentication={this.props.authentication} />
        <div className="container">
          {/* <form> */}
          <ul className="flex-outer">
            <li>
              <label>Drag and drop the archive you with to sign onto this area.</label>
              <FileUpload files={this.state.files || []} onFileAccepted={this.onFileAccepted} />
            </li>
            <li>
              <label>What is the deployment platform this archive is meant for?</label>
              <ul className="flex-inner">
                <li>
                  <input
                    type="radio"
                    id="platform-ios"
                    name="platform"
                    value="ios"
                    onChange={this.onPlatformChanged}
                  />
                  <label htmlFor="platform-ios">iOS</label>
                </li>
                <li>
                  <input
                    type="radio"
                    id="platform-android"
                    name="platform"
                    value="android"
                    onChange={this.onPlatformChanged}
                  />
                  <label htmlFor="platform-android">Android</label>
                </li>
              </ul>
            </li>
            <li>
              <label>You must complete a STRA &amp; PIA before your production release.</label>
              <ul className="flex-inner">
                <li>
                  <input
                    type="checkbox"
                    id="i-agree"
                    name="i-agree"
                    value="agree"
                    onChange={this.onUserDoesAgree}
                  />
                  <label htmlFor="i-agree">Yes, I will</label>
                </li>
              </ul>
            </li>
            <li>
              <button
                onClick={() => {
                  if (!this.validateForm()) {
                    return;
                  }

                  this.props.createSigningJob(this.state.files, this.state.platform);
                }}
              >
                Start
              </button>
            </li>
          </ul>
          {/* </form> */}
          <JobStatusIndicator job={this.props.job} />
        </div>
        <div className="container">
          <Instruction />
        </div>
        <Footer />
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    job: state.job,
    api: state.api,
    authentication: state.authentication,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      createSigningJob,
      login: () => dispatch(authenticateSuccess()),
      logout: () => dispatch(authenticateFailed()),
    },
    dispatch
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
