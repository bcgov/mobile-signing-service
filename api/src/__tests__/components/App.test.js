import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../../components/App';

describe('App Component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <App authentication={{ isAuthenticated: true }} job={{ status: 'testStatus' }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
