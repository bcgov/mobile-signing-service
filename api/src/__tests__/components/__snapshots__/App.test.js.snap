// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <Header
    authentication={
      Object {
        "isAuthenticated": true,
      }
    }
  />
  <div
    className="container"
  >
    <ul
      className="flex-outer"
    >
      <li>
        <label>
          Drag and drop the archive you with to sign onto this area.
        </label>
        <FileUpload
          files={Array []}
          onFileAccepted={[Function]}
        />
      </li>
      <li>
        <label>
          What is the deployment platform this archive is meant for?
        </label>
        <ul
          className="flex-inner"
        >
          <li>
            <input
              id="platform-ios"
              name="platform"
              onChange={[Function]}
              type="radio"
              value="ios"
            />
            <label
              htmlFor="platform-ios"
            >
              iOS
            </label>
          </li>
          <li>
            <input
              id="platform-android"
              name="platform"
              onChange={[Function]}
              type="radio"
              value="android"
            />
            <label
              htmlFor="platform-android"
            >
              Android
            </label>
          </li>
        </ul>
      </li>
      <li>
        <label>
          You must complete a STRA & PIA before your production release.
        </label>
        <ul
          className="flex-inner"
        >
          <li>
            <input
              id="i-agree"
              name="i-agree"
              onChange={[Function]}
              type="checkbox"
              value="agree"
            />
            <label
              htmlFor="i-agree"
            >
              Yes, I will
            </label>
          </li>
        </ul>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Start
        </button>
      </li>
    </ul>
    <JobStatusIndicator
      job={
        Object {
          "status": "testStatus",
        }
      }
    />
  </div>
  <div
    className="container"
  >
    <Instruction />
  </div>
  <Footer />
</div>
`;
